### PR TITLE
Fix updating device list after importing a joint descendant node

### LINKED ISSUE #2476: Cannot import a new device in a robot
Here is the procedure:

1. Import a TouchSensor under a endPoint Solid of HingeJoint node.
2. restart the controller after the import to consider the new device and use the getTouchSensor(tag) to check it.
3. It works only if I use the import function a second time with any other base node.

Here is the robot controller:

```C++
#include <webots/Supervisor.hpp>

// This controller can be tested with dog.wbt

int main() {
  webots::Supervisor *supervisor = new webots::Supervisor();

  if (supervisor->getFromDef("TEST_UNDER_HJ") == NULL) {
    webots::Field* field_under= supervisor->getFromDef("SOLID0")->getField("children");
    field_under->importMFNodeFromString(0, "DEF TEST_UNDER_HJ TouchSensor { name \"test_under_hingejoint\" }");

    /**/
    webots::Field* field = supervisor->getSelf()->getField("children");
    field->importMFNodeFromString(0, "Transform {}");
    /**/

    supervisor->getSelf()->restartController();
  }

  while (supervisor->step(10) != -1) {
    // Path of "touch0" is Robot > HIP0 Hingejoint > Solid > KNEE0 Hingejoint > Solid > KNEESPRING Hingejoint >
    // SOLID0 Solid

    // It works when I start the controller everytime
    webots::TouchSensor* floor_sensor = supervisor->getTouchSensor("touch0");
    printf("touch0: %p\n", floor_sensor);

    // It works only if I import something else (see part: /**/ [..] /**/)
    webots::TouchSensor* floor_sensor_3 = supervisor->getTouchSensor("test_under_hingejoint");
    printf("test_under_hingejoint: %p\n", floor_sensor_3);
  }
  return EXIT_SUCCESS;
}
```

The world file is attached:
[dog.zip](https://github.com/cyberbotics/webots/files/5601414/dog.zip)

### PR DESCRIPTION
**Pull request by stefaniapedrazzi**
Fix #2476: import even in joint is not propagated to the top robot node and the list of devices is not updated.

### CODE REVIEW

**Comment by stefaniapedrazzi** [comment]
Test suite passed successfully, only the upload of the Windows artifact failed but the package was created successfully.

File: src/webots/nodes/WbGroup.cpp [hunk 1]
```diff
@@ -227,17 +232,25 @@ bool WbGroup::isAValidBoundingObject(bool checkOde, bool warning) const {
...
+  if (!parentNode())
+    return;
+
+  WbGroup *pg = dynamic_cast<WbGroup *>(parentNode());
+  if (pg) {
+    pg->descendantNodeInserted(decendant);
+    return;
+  }
+  WbBasicJoint *pj = dynamic_cast<WbBasicJoint *>(parentNode());
+  if (pj) {
```
Comment (omichel) [inline src/webots/nodes/WbGroup.cpp:244]: constness:
```suggestion
  const WbBasicJoint *pj = dynamic_cast<WbBasicJoint *>(parentNode());
  if (pj) {
```
Or maybe simpler like this:
```suggestion
  if (dynamic_cast<WbBasicJoint *>(parentNode())) {
```

File: src/webots/nodes/WbGroup.cpp [hunk 2]
```diff
@@ -227,17 +232,25 @@ bool WbGroup::isAValidBoundingObject(bool checkOde, bool warning) const {
...
+    emit notifyParentJoint(decendant);
+    return;
   }
+  WbSlot *ps = dynamic_cast<WbSlot *>(parentNode());
+  if (ps)
```
Comment (omichel) [inline src/webots/nodes/WbGroup.cpp:249]: Same comment here.

**Review by omichel — DISMISSED** [review]
It looks very good, thank you.
I just have a minor suggestion.

**Review by omichel — APPROVED** [review]
Thank you.

**Pull request merged**